### PR TITLE
[DB-654][risk=no]pagination

### PR DESCRIPTION
**Pull request by mhigbyflowers**

### CODE REVIEW

File: public-ui/package.json [hunk 1]
```diff
@@ -40,6 +40,7 @@
     "immutable": "^4.0.0-rc.9",
     "moment": "^2.20.1",
     "mutationobserver-shim": "^0.3.2",
+    "ngx-pagination": "^4.1.0",
```
Comment (mhigbyflowers) [inline public-ui/package.json:43]: pagination dependency

File: public-ui/src/app/views/ehr-view/ehr-view.component.html [hunk 2]
```diff
@@ -33,9 +33,17 @@ <h5 id="domain-summary" class="secondary-display">
             <div class="results-grid" *ngIf="items.length > 0">
               <div class="domain-info-layout">
                 <span>
-                  <h5 id="domain-name" class="primary-display">
-                    Showing top {{items.length}}
-                  </h5>
+                  <ng-container *ngIf="totalResults <= searchRequest.maxResults; else show_pages">
```
Comment (mhigbyflowers) [inline public-ui/src/app/views/ehr-view/ehr-view.component.html]: if the search is less than max results don't do anything special else format the numbers like (1-50 out of 26,000)

File: public-ui/src/app/views/ehr-view/ehr-view.component.html [hunk 3]
```diff
@@ -126,9 +134,8 @@ <h6 *ngIf="searchText.value" class="primary-display medline-link">Interested in
                 </div>
               </div>
               <div class="tbl-exp-r " [ngClass]="r.expanded?'row-selected':''"
-                *ngFor="let r of standardConcepts.concat(items), let i = index">
+                *ngFor="let r of standardConcepts.concat(items) | paginate: { itemsPerPage: searchRequest.maxResults, currentPage: currentPage, totalItems: totalResults }, let i = index">
```
Comment (mhigbyflowers) [inline public-ui/src/app/views/ehr-view/ehr-view.component.html:137]: pagination setup

File: public-ui/src/app/views/ehr-view/ehr-view.component.html [hunk 4]
```diff
@@ -150,7 +157,11 @@ <h6 *ngIf="searchText.value" class="primary-display medline-link">Interested in
                   <ng-template #code_match_not_standard>
                     <div class="body-lead info-text tbl-d first">
                       <div class="code-tooltip">
-                        <app-highlight-search [indNum]="i+1" [text]="r.conceptName" [searchTerm]="prevSearchText">
+                          <span *ngIf="totalResults > searchRequest.maxResults" class="item-index">
```
Comment (mhigbyflowers) [inline public-ui/src/app/views/ehr-view/ehr-view.component.html:160]: if the search is less than max results don't do anything special else reformat count based on current page

File: public-ui/src/app/views/ehr-view/ehr-view.component.html [hunk 5]
```diff
@@ -285,11 +297,15 @@ <h5 class="no-results"> No results match your search. </h5>
               </div>
             </div>
           </div>
+          <pagination-controls (pageChange)="currentPage = $event;
```
Comment (mhigbyflowers) [inline public-ui/src/app/views/ehr-view/ehr-view.component.html:300]: render pagination pager

File: public-ui/src/app/views/ehr-view/ehr-view.component.ts [hunk 6]
```diff
@@ -139,7 +147,12 @@ export class EhrViewComponent implements OnInit, OnDestroy {
           }
         }));
       this.subscriptions.push(this.searchText.valueChanges
-        .subscribe((query) => localStorage.setItem('searchText', query)));
+        .subscribe((query) => {
```
Comment (mhigbyflowers) [inline public-ui/src/app/views/ehr-view/ehr-view.component.ts:150]: making sure if the user has a space in the search bar that databrowser ignores it

File: public-ui/src/app/views/ehr-view/ehr-view.component.ts [hunk 7]
```diff
@@ -233,12 +246,26 @@ export class EhrViewComponent implements OnInit, OnDestroy {
     } else {
       this.standardConcepts = [];
     }
-    this.top10Results = this.searchResult.items.slice(0, 10);
-    // Set the localStorage to empty so making a new search here does not follow to other pages
-    // localStorage.setItem('searchText', '');
+    // this.top10Results = this.searchResult.items.slice(0, 10);
```
Comment (mhigbyflowers) [inline public-ui/src/app/views/ehr-view/ehr-view.component.ts]: top 10 graph doesn't change with the page

File: public-ui/src/app/views/ehr-view/ehr-view.component.ts [hunk 8]
```diff
@@ -233,12 +246,26 @@ export class EhrViewComponent implements OnInit, OnDestroy {
...
+    this.getTopTen(this.prevSearchText).subscribe((res) => {
+      this.top10Results = res.items.slice(0, 10);
+    });
     this.loading = false;
   }
 
+  public getTopTen(query: string) {
```
Comment (mhigbyflowers) [inline public-ui/src/app/views/ehr-view/ehr-view.component.ts:256]: top 10 graph doesn't change with the page

File: public-ui/src/app/views/ehr-view/ehr-view.component.ts [hunk 9]
```diff
@@ -394,4 +420,10 @@ export class EhrViewComponent implements OnInit, OnDestroy {
     }
     return 10;
   }
+
+  public getNextPage(event) {
```
Comment (mhigbyflowers) [inline public-ui/src/app/views/ehr-view/ehr-view.component.ts:424]: page change event

File: public-ui/src/app/views/ehr-view/ehr-view.component.html (hunk 2 above)
Comment (mhigbyflowers) [inline public-ui/src/app/views/ehr-view/ehr-view.component.html:36]: if the search is less than max results don't do anything special else format the numbers like (1-50 out of 26,000)

**Comment by mhigbyflowers** [comment]
PR quality failing  but
the use of !important is necessary to override dependency styles on the pagination.

**Review by SrushtiGangireddy — APPROVED** [review]
Looks good. thanks.

**Pull request merged**